components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -95,8 +116,18 @@ public void index(Patient patient)
         for (Feature phenotype : patient.getFeatures()) {
             input.addField((phenotype.isPresent() ? "" : "negative_") + phenotype.getType(), phenotype.getId());
         }
+
         input.setField("visibility", this.permissions.getPatientAccess(patient).getVisibility().getName());
         input.setField("accessLevel", this.permissions.getPatientAccess(patient).getVisibility().getPermissiveness());
+
+        // Add genes
+        PatientData<Map<String, String>> allGenes = patient.getData(GENES_KEY);
+        addGenesWithStatusToField(input, allGenes, GENE_STATUS_SOLVED, SOLR_FIELD_SOLVED_GENES);
+        addGenesWithStatusToField(input, allGenes, GENE_STATUS_CANDIDATE, SOLR_FIELD_CANDIDATE_GENES);
+        addGenesWithStatusToField(input, allGenes, GENE_STATUS_REJECTED, SOLR_FIELD_REJECTED_GENES);
+        // Index genes without a status as candidates
+        addGenesWithStatusToField(input, allGenes, null, SOLR_FIELD_CANDIDATE_GENES);
```

Comment:
As discussed with Orion, it may be an empty string not a null (probably if UI is not implemented well or a check  is missed on save?) But in any case from the perspective of this component it is better to have a false positive than to miss a match